utils/storage/storage.js

```diff
@@ -248,10 +251,49 @@ class Storage {
     );
   }
 
+  /**
+   * Function to fetch anonymousId from external source
+   * @param {string} key source of the anonymousId
+   * @returns string
+   */
+  fetchExternalAnonymousId(source) {
+    let anonId;
+    const key = source.toLowerCase();
+    if (!Object.keys(anonymousIdKeyMap).includes(key)) {
+      return anonId;
+    }
+    switch (key) {
+      case "segment":
+        /**
+         * First check the local storage for anonymousId
+         * Ref: https://segment.com/docs/connections/sources/catalog/libraries/website/javascript/#identify
+         */
+        if (Store.enabled) {
+          anonId = Store.get(anonymousIdKeyMap[key]);
+        }
+        if (!anonId && Cookie.IsCookieSupported()) {
+          anonId = Cookie.get(anonymousIdKeyMap[key]);
+        }
+        return anonId;
+
+      default:
+        return anonId;
+    }
+  }
+
   /**
    * get stored anonymous id
    */
-  getAnonymousId() {
+  getAnonymousId(anonymousIdOptions) {
+    const source = get(anonymousIdOptions, "autoCapture.source");
+    if (
+      get(anonymousIdOptions, "autoCapture.enabled") === true &&
+      typeof source === "string"
+    ) {
+      const anonId = this.fetchExternalAnonymousId(source);
+      if (anonId) return anonId;
+    }
+
     return this.parse(
       this.decryptValue(this.storage.get(defaults.user_storage_anonymousId))
```

Comment:
We should first check if the anonymous ID is not available in RS storage and if it is unavailable, then try to retrieve it from the auto-capture source.

Reply:
Addressed